tools/c7n_azure/c7n_azure/resources/key_vault_keys.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.from c7n_azure.provider import resources
+
+import logging
+
+from azure.keyvault.key_vault_id import KeyVaultId
+
+from c7n.filters import Filter
+from c7n.utils import type_schema
+
+from c7n_azure import constants
+from c7n_azure.provider import resources
+from c7n_azure.query import ChildResourceManager, ChildTypeInfo
+from c7n_azure.utils import ThreadHelper, ResourceIdParser
+
+
+log = logging.getLogger('custodian.azure.keyvault.keys')
+
+
+@resources.register('keyvault-keys')
+class KeyVaultKeys(ChildResourceManager):
+
+    class resource_type(ChildTypeInfo):
+        resource = constants.RESOURCE_VAULT
+        service = 'azure.keyvault'
+        client = 'KeyVaultClient'
+        enum_spec = (None, 'get_keys', None)
+
+        parent_manager_name = 'keyvault'
+        raise_on_exception = False
+
+        @classmethod
+        def extra_args(cls, parent_resource):
+            return {'vault_base_url': 'https://{0}.vault.azure.net'.format(parent_resource['name'])}
+
+
+@KeyVaultKeys.filter_registry.register('keyvault')
+class KeyvaultFilter(Filter):
+    schema = type_schema(
+        'keyvault',
+        required=['vaults'],
+        **{
+            'vaults': {'type': 'array', 'items': {'type': 'string'}}
+        }
+    )
+
+    def process(self, resources, event=None):
+        return [r for r in resources
+                if ResourceIdParser.get_resource_name(r['c7n:parent-id']) in self.data['vaults']]

```

Comment:
Let's use a constant for 'c7n:parent-id' will be reused for other child resources.